File: WikiAnalyser/FootballMatchAnalyser/FootballStatElement.py
class FootballStatElement:
	"""Gesamtstatistik für einen Gegner"""

	Name = ""
	Flag = ""
	Year = 0
	Matches = []
	PreComment = "";
	PostComment = "";
	
	@property
	def Played(self):
		return self.Win + self.Drawn + self.Lose

	Win = 0
	Drawn = 0
	Lose = 0
	GoalsFor = 0
	GoalsAgainst = 0
	GoalsDIff = 0

	@property
	def GoalsDiff(self):
		return self.GoalsFor - self.GoalsAgainst

	@property
	def Points(self):
		return self.Win * 3 + self.Drawn

	@property
	def Balance(self):
		return self.Win - self.Lose

	@property
	def ColorCode(self):
		return "CCFFCC" if Balance > 0 else "FFCCCC" if Balance < 0 else "FFFFCC"

	def AddMatch(self, match):
		if match.Result != "X":
			self.Matches.append(match)
			if match.HomeTeam == match.OpponentTeam:
				if match.ResultHome < match.ResultAway:
					self.Win += 1
				elif match.ResultHome > match.ResultAway:
					self.Lose += 1
				else:
					self.Drawn += 1
				self.GoalsFor += match.ResultAway;
				self.GoalsAgainst += match.ResultHome
			else:
				if match.ResultHome > match.ResultAway:
					self.Win += 1
				elif match.ResultHome < match.ResultAway:
					self.Lose += 1
				else:
					self.Drawn += 1
				self.GoalsFor += match.ResultHome;
				self.GoalsAgainst += match.ResultAway
		
	@property
	def OpponentWikicode (self):
		return "|- style=\"background:#{0};\"\n| style=\"text-align:left;\" | {1}\n| {2} || {3} || {4} || {5} || {6} || {7} || {8:+0;-0;+0} || {9}".format(
			self.ColorCode, self.Flag, self.Played, self.Win, self.Drawn, self.Lose, self.GoalsFor, self.GoalsAgainst, self.GoalDiff, self.Points)

	@property
	def YearWikicode (self):
		return "|-\n| '''{0}''' || {1} || {2} || {3} || {4} || {5} || {6} || {7:+0;-0;+0} || {8}".format(
			self.Year, self.Played, self.Win, self.Drawn, self.Lose, self.GoalsFor, self.GoalsAgainst, self.GoalDiff, self.Points)

	def __init__(self, name, flag):
		"""
		Erstellt neues FootballStatElement
		name (String): Staatsname
		flag (String): Flaggenkürzel
		"""
		self.Name = name
		self.Flag = str.format("{{{{{0}|#}}}}", flag)
		self.Matches = []

	def __init__(self, year):
		"""
		Erstellt neues FootballStatElement
		year (int): Jahr
		"""
		self.Year = year
		self.Matches = []

File: PwrBot/FootballStatElement.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;

namespace Simocracy.PwrBot
{


	/// <summary>
	/// Gesamtstatistik für einen Gegner
	/// </summary>
	[DebuggerDisplay("Name={Name}, Flag={Flag}, Matches={Matches.Count}, WDL={Win}-{Drawn}-{Lose}, Goals={GoalsFor}-{GoalsAgainst}")]
	public class FootballStatElement
	{
		/// <summary>
		/// Staatsname
		/// </summary>
		public string Name { get; set; } = String.Empty;
		public string Flag { get; set; } = String.Empty;
		/// <summary>
		/// Staatsname
		/// </summary>
		public int Year { get; set; } = 0;
		/// <summary>
		/// Matchliste
		/// </summary>
		public List<FootballMatch> Matches { get; set; } = new List<FootballMatch>();
		/// <summary>
		/// Kommentar am Anfang
		/// </summary>
		public string PreComment { get; set; } = String.Empty;
		/// <summary>
		/// Kommentar am Ende
		/// </summary>
		public string PostComment { get; set; } = String.Empty;

		public int Played => Win + Drawn + Lose;
		public int Win { get; private set; }
		public int Drawn { get; private set; }
		public int Lose { get; private set; }
		public int GoalsFor { get; private set; }
		public int GoalsAgainst { get; private set; }
		public int GoalDiff => GoalsFor - GoalsAgainst;
		public int Points => Win * 3 + Drawn;

		public int Balance => Win - Lose;

		/// <summary>
		/// Hex-Code der Farbe basierend auf <see cref="Balance"/>, ohne führendes #!
		/// </summary>
		public string ColorCode => Balance > 0 ? "CCFFCC" : Balance < 0 ? "FFCCCC" : "FFFFCC";

		/// <summary>
		/// Wikicode für Gegnertabelle
		/// </summary>
		public string OpponentWikicode =>
			$"|- style=\"background:#{ColorCode};\"\n| style=\"text-align:left;\" | {Flag}\n| {Played} || {Win} || {Drawn} || {Lose} || {GoalsFor} || {GoalsAgainst} || {GoalDiff:+0;-0;+0} || {Points}"
			;

		/// <summary>
		/// Wikicode für Jahrestabelle
		/// </summary>
		public string YearWikicode =>
			$"|-\n| '''{Year}''' || {Played} || {Win} || {Drawn} || {Lose} || {GoalsFor} || {GoalsAgainst} || {GoalDiff:+0;-0;+0} || {Points}"
			;

		public void AddMatch(FootballMatch match)
		{
			if(match.Result != "X")
			{
				Matches.Add(match);
				if(match.HomeTeam == match.OpponentTeam)
				{
					if(match.ResultHome < match.ResultAway)
						Win++;
					else if(match.ResultHome > match.ResultAway)
						Lose++;
					else
						Drawn++;

					GoalsFor += match.ResultAway;
					GoalsAgainst += match.ResultHome;
				}
				else
				{
					if(match.ResultHome > match.ResultAway)
						Win++;
					else if(match.ResultHome < match.ResultAway)
						Lose++;
					else
						Drawn++;

					GoalsFor += match.ResultHome;
					GoalsAgainst += match.ResultAway;
				}
			}
		}
	}
}
